euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/ReduceStateByKey.java

```diff
@@ -32,26 +32,60 @@
 import java.util.Objects;
 
 /**
- * Operator reducing state by given reduce function.
+ * A {@link ReduceStateByKey} operator is a stateful, complex, lower-level-api,
+ * but very powerful processor and serves as the basis for easier-to-use
+ * operators. Client API users have generally little motivation to use it
+ * directly and should prefer {@link ReduceByKey} when possible.<p>
  *
- * @param <IN>     Type of input records
- * @param <KIN>    Type of records entering #keyBy and #valueBy methods
- * @param <WIN>    Type of input records being windowed
- * @param <KEY>    Output type of #keyBy method
- * @param <VALUE>  Output type of #valueBy method
- * @param <KEYOUT> Type of output key
- * @param <OUT>    Type of output value
+ * The operator assigns each input item to a set of windows (through a user
+ * provided {@link Windowing} implementation) and turns the item into a
+ * key/value pair. For each of the assigned windows the extracted value is
+ * accumulated using a user provided {@link StateFactory state implementation}
+ * under the extracted key. I.e. the value is accumulated into a state
```

Comment:
Just a detail. I wouldn't start a new sentence with `I.e.`. It looks really weird.